MdeModulePkg/Core/Dxe/Gcd/Gcd.c

```diff
@@ -2615,6 +2621,10 @@ CoreInitializeGcdServices (
             GcdMemoryType = EfiGcdMemoryTypePersistent;
           }
 
+          if ((ResourceHob->ResourceAttribute & EFI_RESOURCE_ATTRIBUTE_SPECIAL_PURPOSE) == EFI_RESOURCE_ATTRIBUTE_SPECIAL_PURPOSE) {

```

Comment:
This looks ok to me as long as you don't expect special purpose memory to be persistent or reserved. The PI spec does not seem to indicate anything about these being mutually exclusive, but I can understand how if a resource descriptor HOB reports a special purpose region that that should take precedence over any other region. Is this your expectation @Patrick-Payne?

Reply:
I spoke with subject matter experts, for this particular routine, if both EFI_RESOURCE_ATTRIBUTE_SPECIAL_PURPOSE and EFI_RESOURCE_ATTRIBUTE_PERSISTENT flags are set, the persistent behavior takes priority. I've changed the code accordingly.